skimage/transform/radon_transform.py

```diff
@@ -110,7 +110,8 @@ def radon(image, theta=None, circle=True, *, preserve_range=None):
     if padded_image.shape[0] != padded_image.shape[1]:
         raise ValueError('padded_image must be a square')
     center = padded_image.shape[0] // 2
-    radon_image = np.zeros((padded_image.shape[0], len(theta)))
+    radon_image = np.zeros((padded_image.shape[0], len(theta)),
+                           dtype=image.dtype)
 
     for i, angle in enumerate(np.deg2rad(theta)):
```

Comment:
does theta need to be declared as single as well in this case?

Reply:
Not necessarily: in `warp`,  only `padded_image` dtype acts on output dtype, but for consistency, the rotation matrix `R` can be cast to dtype. ;-)